src/ext/framework/help_commands.rs

```diff
@@ -0,0 +1,183 @@
+                .description("To get help about individual command, pass its name as an argument \
+                              to this command.");
+            for (name, group) in groups {
+                let mut desc = String::new();
+                if let Some(ref x) = group.prefix {
+                    desc.push_str(&format!("Prefix: {}\n", x));
+                }
+                let mut no_commands = true;
+                desc.push_str("Commands:\n");
+                for (n, cmd) in &group.commands {
+                    if cmd.help_available {
+                        desc.push_str(&format!("`{}`\n", n));
+                        no_commands = false;
+                    }
+                }
+                if no_commands {
+                    desc.push_str("*[No commands]*");
+                }
+                embed = embed.field(|f| {
+                    f.name(&name)
+                        .value(&desc)
+                });
+            }
+            embed
+        })
+    });
+}
+
+pub fn plain(ctx: &Context,
+             _: &Message,
+             groups: HashMap<String, Arc<CommandGroup>>,
+             args: Vec<String>) {
+    if args.len() > 0 {
+        let name = args.join(" ");
+        for (_, group) in groups {
+            let mut found: Option<(&String, &Command)> = None;
+            if let Some(ref prefix) = group.prefix {
+                for (command_name, command) in &group.commands {
+                    if name == format!("{} {}", prefix, command_name) {
+                        found = Some((command_name, command));
+                    }
+                }
+            } else {
+                for (command_name, command) in &group.commands {
+                    if name == command_name[..] {
+                        found = Some((command_name, command));
+                    }
+                }
+            };
+            if let Some((command_name, command)) = found {
+                if !command.help_available {
+                    let _ = ctx.say("**Error**: No help available.");
+                    return;
+                }
+                let mut result = format!("**{}**\n", command_name);
+                if let Some(ref desc) = command.desc {
+                    result.push_str(&format!("**Description:** {}", desc));
+                }
+                if let Some(ref usage) = command.usage {
+                    result.push_str(&format!("**Usage:** {}", usage));
+                }
+                result.push_str(&format!("**Available:** {}",
+                                         if command.dm_only {
+                                             "Only in DM"
+                                         } else if command.guild_only {
+                                             "Only in guilds"
+                                         } else {
+                                             "Everywhere"
+                                         }));
```

Comment:
Maybe the return value of this if-elseif-else can be bound, and then this can be `write!(result, "**Available**: {}", location);`